File: src/main/java/org/unbrokendome/gradle/plugins/testsets/internal/PredefinedUnitTestSet.java
package org.unbrokendome.gradle.plugins.testsets.internal;

import java.util.Collections;
import java.util.Set;

import org.gradle.api.plugins.JavaPlugin;
import org.gradle.api.tasks.SourceSet;
import org.unbrokendome.gradle.plugins.testsets.dsl.TestSet;

public class PredefinedUnitTestSet extends AbstractTestSet {

	public static final String NAME = "unitTest";


	@Override
	public String getName() {
		return NAME;
	}
	

	@Override
	public boolean isCreateArtifact() {
		return false;
	}

	
	@Override
	public String getDirName() {
		return SourceSet.TEST_SOURCE_SET_NAME;
	}
	

	@Override
	public Set<TestSet> getExtendsFrom() {
		return Collections.emptySet();
	}


	@Override
	public String getTestTaskName() {
		return JavaPlugin.TEST_TASK_NAME;
	}


	@Override
	public String getSourceSetName() {
		return SourceSet.TEST_SOURCE_SET_NAME;
	}


	@Override
	public String getCompileConfigurationName() {
		return JavaPlugin.TEST_COMPILE_CONFIGURATION_NAME;
	}


	@Override
	public String getRuntimeConfigurationName() {
		return JavaPlugin.TEST_RUNTIME_CONFIGURATION_NAME;
	}
}

File: src/main/kotlin/org/unbrokendome/gradle/plugins/testsets/dsl/PredefinedUnitTestSet.kt
package org.unbrokendome.gradle.plugins.testsets.dsl

import org.gradle.api.plugins.JavaPlugin
import org.gradle.api.tasks.SourceSet
import org.gradle.testing.jacoco.plugins.JacocoPlugin


internal class PredefinedUnitTestSet(container: TestSetContainer, sourceSet: SourceSet)
    : AbstractTestSetBase(container, "unitTest", sourceSet), TestSet {

    override val testTaskName: String
        get() = JavaPlugin.TEST_TASK_NAME


    override val jarTaskName: String
        get() = "testJar"


    override var classifier: String = "tests"


    override var environment: Map<String, Any> = mutableMapOf()
        set(value) {
            field = value
            notifyObservers { it.environmentVariablesChanged(this, value) }
        }


    override var systemProperties: Map<String, Any?> = mutableMapOf()
        set(value) {
            field = value
            notifyObservers { it.systemPropertiesChanged(this, value) }
        }
}
